### PR TITLE
Fix slow performance with S3 #662 #301 #508

### PR DESCRIPTION
**Pull request by krukas**
We encounter the same problem as #662 #301 #508, and had the same solutions as others have suggested.

### CODE REVIEW

**Comment by khink** [comment]
This would be nice to have.

**Comment by krukas** [comment]
What do I need to do to get this merged? @uri-rodberg @JanMalte

File: sorl/thumbnail/images.py [hunk 1]
```diff
@@ -18,6 +18,14 @@
...
 url_pat = re.compile(r'^(https?|ftp):\/\/')
 
+STORAGES_CACHE = {}
+
+
+def get_or_create_storage(storage):
+    if storage not in STORAGES_CACHE:
+        STORAGES_CACHE[storage] = get_module_class(storage)()
+    return STORAGES_CACHE[storage]
+
```
Comment (claudep) [inline sorl/thumbnail/images.py:28]: What about using https://docs.python.org/3/library/functools.html#functools.lru_cache instead?

**Review by claudep — CHANGES_REQUESTED** [review]

File: sorl/thumbnail/images.py (hunk 1 above)
Comment (krukas) [reply sorl/thumbnail/images.py:26]: Thanks for the suggestion, I have updated it to use `lru_cache`

**Review by claudep — APPROVED** [review]
Thanks, looks good now!

**Pull request merged**